Makefile

```diff
@@ -92,7 +92,9 @@ endif
 build: ensure-submodule-initialized buf build-module
 	cd viam-cartographer && cmake -Bbuild -G Ninja ${EXTRA_CMAKE_FLAGS} && cmake --build build
 
-build-debug: EXTRA_CMAKE_FLAGS += -DCMAKE_BUILD_TYPE=Debug -DFORCE_DEBUG_BUILD=True
+build-debug: ASAN_OPTIONS="detect_leaks=1 detect_stack_use_after_return=true"
```

Comment:
This adds address sanitizer to debug builds. This is how I tracked down what was causing the segfaults on mac where I can't run valgrind.